blocks/src/custom-status/block.js

```diff
@@ -11,19 +11,37 @@ let { SelectControl } = wp.components;
 /**
  * Map Custom Statuses as options for SelectControl
  */
-let statuses = window.EditFlowCustomStatuses.map( s => ({ label: s.name, value: s.slug }) );
+let customStatuses = window.EditFlowStatuses.custom_statuses.map( s => ({ label: s.name, value: s.slug }) );
 
-let getStatusLabel = slug => statuses.find( s => s.value === slug ).label;
+// Trash isn't in `default_statuses` but we need to check for it
+let isCoreStatus = slug => window.EditFlowStatuses.default_statuses.find( s => s.slug === slug ) || slug === 'trash';
+let getCustomStatusLabel = slug => customStatuses.find( s => s.value === slug ).label;
 
 // Hack :(
 // @see https://github.com/WordPress/gutenberg/issues/3144
-let sideEffectL10nManipulation = status => {
-  let node = document.querySelector('.editor-post-save-draft');
-  if ( node ) {
-    document.querySelector( '.editor-post-save-draft' ).innerText = `${ __( 'Save' ) }`
-  }
+// Gutenberg will also override the status set in '.editor-post-save-draft' after save, and there isn't yet a way
+// to subscribe to a "post save" message. So instead set a timeout and override the text in '.editor-post-save-draft'
+let schedulePostStatusUpdater = () => {
+  setTimeout(() => {
```

Comment:
Potentially a more "native" way of doing this would be to monitor for changes to the modified time (set modified time on load and compare/update/fire action in `wp.data.subscribe` callback.

Reply:
Yea, not a bad thought, I'll mess around and see if I can get that working

Reply:
Though I'm not sure it'll solve the underlying issue. We can observe [post update](https://github.com/Automattic/Edit-Flow/blob/master/blocks/src/custom-status/block.js#L89), but the problem is [here](https://github.com/WordPress/gutenberg/blob/95e769df1f82f6b0ef587d81af65dd2f48cd1c38/packages/editor/src/components/post-saved-state/index.js#L38-L41).

Gutenberg is setting a timeout of 1s to flip the message from "Saving..." to the [original text](https://github.com/WordPress/gutenberg/blob/95e769df1f82f6b0ef587d81af65dd2f48cd1c38/packages/editor/src/components/post-saved-state/index.js#L104): "Save as `Draft|Pending`". But if there's concern over about polling, we could set one timeout for something like 1.2s and that might solve it (but small race condition)

Reply:
I get what you mean. The two options you proposed (callback loop and timeout after save) make sense to me based on current Gutenberg code. This seems like it could also be a good place for an upstream change to Gutenberg if there's a filter that could make it work. Best idea I have right now would be providing a way to filter/bring your own function to use instead of `isPending ? __( 'Save as Pending' ) : __( 'Save Draft' );`